### ISSUE TITLE
Add extra validations to launch templates

### ISSUE THREAD

**Issue by monwolf** [aws-cloudformation/cfn-lint#2611]
When you create a launch template as follows:

```
  ProxyLaunchTemplate: 
    Type: "AWS::EC2::LaunchTemplate"
    Properties:
      LaunchTemplateName: !Sub '${AccountName}-proxy-template-amd64'
      LaunchTemplateData:
        KeyName: !Ref KeyName
        ImageId: !Ref AmazonLinuxAMI
        IamInstanceProfile:
          Arn: !GetAtt 'ProxyInstanceProfile.Arn' 
        NetworkInterfaces:
        - DeleteOnTermination: true
          Description: Primary network interface
          DeviceIndex: '0'
          AssociatePublicIpAddress: true
        SecurityGroupIds: 
          - !Ref ProxySG
        InstanceType: !Ref InstanceType
        BlockDeviceMappings: 
          - DeviceName: "/dev/xvda"
            Ebs: 
              VolumeSize: "8"
              VolumeType: "gp2"
```

There's an error on CF saying the SG must be specified on Network interface but the cfn-lint didn't catch it. The same occur when you define the network interface with a SG and the groups field is specified inside NetworkInterfaces:

```
        NetworkInterfaces:
        - DeleteOnTermination: true
          Description: Primary network interface
          DeviceIndex: '0'
          AssociatePublicIpAddress: true
          Groups:
            - !Ref ProxySG
```

So if the validation should be if NetworkInterfaces extists SGs must be specified inside but not at LaunchTemplate level, if not SGs should be specified at LaunchTemplate Properties

**Comment by kddejong** [comment]
This was a lot easier to write into the v1 version of cfn-lint.  I'm going to keep this issue open until v1 is officially released.
